File: apple/Peacock/Protocol/PacketType.swift
import Foundation

enum PacketType: UInt16, Sendable {
    case announce = 1
    case bye = 2
    case announceResponse = 3
    case text = 10
    case fileOffer = 20
    case fileAccept = 21
    case fileReject = 22
    case fileChunk = 23
    case clipboard = 30
    case snippetShare = 31
    case ack = 99
}

File: android/app/src/main/java/com/peacock/app/protocol/PacketType.kt
package com.peacock.app.protocol

enum class PacketType(val value: UShort) {
    Announce(1u),
    Bye(2u),
    AnnounceResponse(3u),
    Text(10u),
    FileOffer(20u),
    FileAccept(21u),
    FileReject(22u),
    FileChunk(23u),
    Clipboard(30u),
    SnippetShare(31u),
    Ack(99u);

    companion object {
        fun fromValue(v: UShort): PacketType? = entries.find { it.value == v }
    }
}
